PWGHF/TableProducer/candidateCreator3Prong.cxx

```diff
@@ -235,25 +274,135 @@ struct HfCandidateCreator3Prong {
     }
   }
 
+  ///////////////////////////////////
+  ///                             ///
+  ///   No centrality selection   ///
+  ///                             ///
+  ///////////////////////////////////
+
+  /// @brief process function w/ PV refit and w/o centrality selections
   void processPvRefit(aod::Collisions const& collisions,
                       soa::Join<aod::Hf3Prongs, aod::HfPvRefit3Prong> const& rowsTrackIndexProng3,
                       aod::TracksWCovExtra const& tracks,
                       aod::BCsWithTimestamps const& bcWithTimeStamps)
   {
-    runCreator3Prong<true>(collisions, rowsTrackIndexProng3, tracks, bcWithTimeStamps);
+    runCreator3Prong</*doPvRefit*/ true, CentralityEstimator::None>(collisions, rowsTrackIndexProng3, tracks, bcWithTimeStamps);
   }
+  PROCESS_SWITCH(HfCandidateCreator3Prong, processPvRefit, "Run candidate creator with PV refit and w/o centrality selections", false);
 
-  PROCESS_SWITCH(HfCandidateCreator3Prong, processPvRefit, "Run candidate creator with PV refit", false);
-
+  /// @brief process function w/o PV refit and w/o centrality selections
   void processNoPvRefit(aod::Collisions const& collisions,
                         aod::Hf3Prongs const& rowsTrackIndexProng3,
                         aod::TracksWCovExtra const& tracks,
                         aod::BCsWithTimestamps const& bcWithTimeStamps)
   {
-    runCreator3Prong(collisions, rowsTrackIndexProng3, tracks, bcWithTimeStamps);
+    runCreator3Prong</*doPvRefit*/ false, CentralityEstimator::None>(collisions, rowsTrackIndexProng3, tracks, bcWithTimeStamps);
+  }
+  PROCESS_SWITCH(HfCandidateCreator3Prong, processNoPvRefit, "Run candidate creator without PV refit and w/o centrality selections", true);
+
+  /////////////////////////////////////////////
+  ///                                       ///
+  ///   with centrality selection on FT0A   ///
+  ///                                       ///
+  /////////////////////////////////////////////
+
+  /// @brief process function w/ PV refit and w/ centrality selection on FT0A
+  void processPvRefitCentFT0A(soa::Join<aod::Collisions, aod::CentFT0As> const& collisions,
+                              soa::Join<aod::Hf3Prongs, aod::HfPvRefit3Prong> const& rowsTrackIndexProng3,
+                              aod::TracksWCovExtra const& tracks,
+                              aod::BCsWithTimestamps const& bcWithTimeStamps)
+  {
+    runCreator3Prong</*doPvRefit*/ true, CentralityEstimator::FT0A>(collisions, rowsTrackIndexProng3, tracks, bcWithTimeStamps);
   }
+  PROCESS_SWITCH(HfCandidateCreator3Prong, processPvRefitCentFT0A, "Run candidate creator with PV refit and w/ centrality selection on FT0A", false);
```

Comment:
I would not define different process functions for the different centrality estimators. To keep the entropy low, an approach similar to the v2 task can be adopted https://github.com/AliceO2Group/O2Physics/blob/d2f45f354bdaf98d9ff0f8c35da0a7d37dca95d3/PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx#L72
where we keep all the possible centrality estimators together. Maybe we could keep the options with and w/o centrality, so that the centrality task can be removed from the workflow if not needed.
What we lose is the possibility to disentangle the centrality estimator at compile time as it is done at the moment. But the performance impact should not be large.

Reply:
I would prefer to have it split as implemented by @mfaggin, to avoid the increase of the memory when not needed. In fact, the centrality tables are produced only if consumed by later workflows. One intermediate solution could be to reduce the number of estimators here, and keep only `FT0C` and `FT0M` for the moment.

Reply:
Fine for me